e2e-tests/cypress/support/page-objects/Discover.ts

```diff
@@ -83,6 +86,76 @@ export class Discover {
     }
     return openDataPageForKibanaBefore7_18_1();
   };
+
+  static navigateTo() {
+    cy.log('navigateTo');
+    KibanaNavigation.openKibanaNavigation();
+    cy.contains('Discover').click({force: true});
+    cy.wait(1000);
+  }
+
+  static selectDataView(dataViewName: string) {
+    cy.log('selectDataView');
+
+    // For Kibana 8.0.0 and above, we need to interact with the popover
+    if (semver.gte(getKibanaVersion(), '8.0.0')) {
+      cy.get('[data-test-subj="discover-dataView-switch-link"]').click(); // Open the popover
+
+      // Search for the data view within the popover
+      cy.get('.euiPopover__panel [role="combobox"] input') // Assuming this is the search input within the popover
+        .type(dataViewName);
+
+      // Wait for the options to filter and then select the desired data view
+      cy.wait(500); // Adjust if needed
+      cy.get(`.euiPopover__panel li:contains("${ dataViewName }")`)
+        .click({force: true});
+    } else {
+      // For older Kibana versions, use the previous logic
+      cy.get('[data-test-subj="indexPattern-switch-link"]').click();
+      cy.contains(dataViewName).click({force: true});
+    }
+  }
+
+  static generateCsvReport(timeout: number = 50000) {
```

Comment:
currently, we know nothing about the timeout unit. When you use a type that doesn't support it, it's a good practise to add a hint in the param name e.g. `timeoutSeconds`

Reply:
Deleted